src/main/java/dk/kb/datahandler/util/HarvestTimeUtil.java

```diff
@@ -39,7 +47,12 @@ public static synchronized String loadLastHarvestTime(OaiTargetDto oaiTarget) th
     
     public static synchronized void updateDatestampForOaiTarget(OaiTargetDto oaiTarget, String datestamp) throws Exception{        
         
-        if (!validateDataFormat(datestamp)) {
+        //Hack to fix datestamp return from Preservica. Format returned are not in OAI standard and is parsed wrong (downgrade to day) when given to preservica.
+        if (datestamp.length() >20) { // 2021-03-24T19:57:34.123Z -> 2021-03-24T19:57:34Z 
+            datestamp=datestamp.substring(0,19)+"Z";
+        }
+            
+        if (!validateOaiDateFormat(datestamp)) {
```

Comment:
This fails if the server returns a simple and valid `YYYY-mm-dd` as `datestamp`. Please handle both day- and second-granularity when validating.

Reply:
Fixed for the yyyy-MM-dd format. But I doubt it can ever be returned as dateStamp as last record time, since that
day will then be harvested again. Actually forever by the batch-job every time it starts.